app/javascript/article-form/components/ImageUploader.jsx

```diff
@@ -349,11 +366,14 @@ export const ImageUploader = ({
         });
         break;
       case 'success':
-        onImageUploadSuccess?.(`![Image description](${message.link})`);
-        dispatch({
-          type: 'upload_image_success',
-          payload: { insertionImageUrls: [message.link] },
-        });
+        if (message?.links?.length > 0) {
+          formatUploadedImages(message?.links)
+
+          dispatch({
+            type: 'upload_image_success',
+            payload: { insertionImageUrls: [message.link] },
```

Comment:
@narender2031 It seems isn't quite right here, but I'm a _little_ uncertain what the right solution might be. It seems this currently assumes that _all_ messages might contain `links`, but the test scenarios, at least, are all passing `link` (singular, no "s"). 

It's possible that this is now an issue in the test — maybe switching this input to `multiple` will send plural `message.links` regardless of how many images are being uploaded? If changing the input type to `multiple` means _always_ getting `links` (plural, plus "s"), then the tests need to be updated with that language.

If `multiple` input still sends singular `link` for a single image, then the code here will need to deal with that possibility, something like:

```suggestion
        const links = (message.link) ? [message.link] : message.links

        if (links.length > 0) {
          formatUploadedImages(links)

          dispatch({
            type: 'upload_image_success',
            payload: { insertionImageUrls: links },
```